p2p/upgrade.go

```diff
@@ -63,6 +64,23 @@ type Host struct {
 	bootstrap *bootstrap.Bootstrap
 }
 
+func isBootnode(h host.Host, bootnodes []string) (bool, error) {
+	best, err := peerexchange.BestHostAddress(h)
```

Comment:
there is an h.ID, best is needed only for discovery

Reply:
done